src/app/data-planes/sources.ts

```diff
@@ -23,12 +23,126 @@ export type DataplaneRulesCollectionSource = DataSourceResponse<DataplaneRulesCo
+      try {
+        return JSON.parse(val)
+      } catch (e) {
+        return val
+      }
+    })(value.join(':').trim())
+
+    // walk the path creating `json.objects: value`
+    key.split('.').reduce((prev, item, i, arr) => {
+      if (i === arr.length - 1) {
+        // if this is the last segment in the path
+        // then just set the value
+        prev[item] = val
+        return prev
+      }
+      // otherwise add a {} if its not there already
+      // and return that as prev so we can continue appending
+      if (typeof prev[item] === 'undefined') {
+        const val = {}
+        prev[item] = val
+        return val
+      } else {
+        return prev[item]
+      }
+    }, prev)
+
+    return prev
+  }, {} as MetricRecord)
+}
+
+// lets you specify the things you are interested in
+const getTraffic = (json: ReturnType<typeof parse>, filter: (key: string) => boolean, protocols = ['http', 'tcp']) => {
+  return protocols.map((item) => {
+    return Object.entries(json[item]).filter(([key, _value]) => {
+      return filter(key)
+    }).map(([key, value]) => {
+      const parts = key.split('_')
+      // check see if the last thing is a number,
+      // if so its probably a port
+      const last = parts.pop()
+
+      let port
+      if (last && isNaN(parseInt(last))) {
+        // parts.push(last)
+      } else {
+        port = last
+      }
+      parts.push(last)
+      return {
+        metrics: value,
+        protocol: item,
+        name: parts.join('_'),
+        ...(port ? { port } : {}),
+      }
+    })
+  }).reduce((prev, item) => prev.concat(item))
+}
 
 export const sources = (api: KumaApi) => {
   return defineSources({
     '/meshes/:mesh/dataplanes/:name': async (params) => {
       return api.getDataplaneFromMesh(params)
     },
+    '/meshes/:mesh/dataplanes/:name/traffic': async (params) => {
+      const { mesh, name } = params
+
+      const res = await api.getDataplaneData({
+        mesh,
+        dppName: name,
+        dataPath: 'stats',
+      })
+
+      // parse the stuff
+      const json = parse(res)
+
+      // inbounds is the thing in `cluster.` starting with `localhost_`
+      const inbounds = getTraffic(json, (key) => key.startsWith('localhost_'), ['cluster'])
+
+      // outbounds are anything else unless it starts with something in the below list
+      const outbounds = getTraffic(json, (key) => {
+        return ![
+          'admin',
+          'async-client',
+          'kuma_envoy_admin',
+          'probe_listener',
+          'localhost_',
+          'inbound_passthrough',
+          'outbound_passthrough',
+        ].some(item => key.startsWith(item))
+      })
+
+      // passthrough traffic is anything in `http.` or `tcp.` that starts with this list
+      const passthrough = getTraffic(json, (key) => [
+        'outbound_passthrough',
+      ].some(item => key.startsWith(item)))
+
+      // see what you are doing
+      console.log(passthrough, inbounds, outbounds)
```

Comment:
From the JSON structure, pick out the things we need.

Not entirely sure whether we need to merge more things, like I think it looks like passthrough could be `ipv6` and/or `ipv4` so I guess we need to merge/aggregate those?